### ISSUE TITLE
React 16.13.X warning when Field has validator

### ISSUE THREAD

**Issue by yann-combarnous** [final-form/react-final-form#768]
### Are you submitting a **bug report** or a **feature request**?

Bug report

### What is the current behavior?

https://codesandbox.io/s/react-1613x-final-form-array-yfcmx

Instructions:
* Click on "Show username" checkbox
* Watch console error

```
Warning: Cannot update a component (`ReactFinalForm`) while rendering a different component (`Field`). To locate the bad setState() call inside `Field`, follow the stack trace as described in https://fb.me/setstate-in-render
    in Field (at MyFieldArray.js:12)
    in FieldArray (at MyFieldArray.js:6)
    in MyFieldArray (at App.js:30)
    in form (at App.js:20)
    in ReactFinalForm (at App.js:12)
    in div (at App.js:11)
    in App (at src/index.js:7)
```

### What is the expected behavior?

No console warning

### What's your environment?

Latest React final form 6.4.0 and latest final form library 4.19, see codesandbox

**Comment by yann-combarnous** [comment]
See offending code at https://github.com/final-form/react-final-form/issues/751#issuecomment-606725324

**Comment by erikras** [comment]
Published fix in [`final-form@4.19.1`](https://github.com/final-form/final-form/releases/tag/v4.19.1).

**Comment by zaguiini** [comment]
The problem is happening on `4.20.1`. Was the fix reverted by any chance?
